### ISSUE TITLE
Document distributed training and the DataAccessor class

### ISSUE THREAD

**Issue by cattuna** [tensorflow/tfx#4792]
All of [tutorial pages](https://www.tensorflow.org/tfx/tutorials) cites using the `DataAcessor` class, something like this:

```
  return data_accessor.tf_dataset_factory(
      file_pattern,
      tfxio.TensorFlowDatasetOptions(
          batch_size=batch_size, label_key=_LABEL_KEY),
      schema=schema).repeat()
```

It's not clear to me what this does and whether if it has built in support for distributed/multi-worker training. There's a gap between Tensorflow's [distributed training tutorials](https://www.tensorflow.org/tutorials/distribute/keras) and the usage of this class. 

For example, some of the questions I have: 

* Do I need to use [`tf.distribute.Strategy.experimental_distribute_dataset`](https://www.tensorflow.org/api_docs/python/tf/distribute/Strategy#experimental_distribute_dataset) or [`tf.distribute.Strategy.distribute_datasets_from_function`](https://www.tensorflow.org/api_docs/python/tf/distribute/Strategy#distribute_datasets_from_function) to create a [`tf.distribute.DistributedDataset`](https://www.tensorflow.org/api_docs/python/tf/distribute/DistributedDataset)? If so, do I do this after mapping the transform layer or before?
* How to set `batch_size` in relation to the number of replicas? 
* Do I need to specify `tf.data.Options()` when using this class?

All and all, my primary hope is that there's more end to end examples in relation to distributed training. None of the[ tutorial pages](https://www.tensorflow.org/tfx/tutorials) use multiple-worker or more than 1 GPUs.

**Comment by bli00** [comment]
@pindinagesh any updates on this?

**Comment by singhniraj08** [comment]
@singhniraj08,

[tfx.v1.components.DataAccessor](https://www.tensorflow.org/tfx/api_docs/python/tfx/v1/components/DataAccessor) contains factories that can create tf.data.Datasets or other means to access the train/eval data on disk. The function mentioned by you is used to generate features and label for training.

For distributed training, You can follow [Vertex AI Training and Serving with TFX and Vertex Pipelines](https://www.tensorflow.org/tfx/tutorials/tfx/gcp/vertex_pipelines_vertex_training#run_the_pipeline_using_a_gpu) which guides on [running pipeline using GPU](https://www.tensorflow.org/tfx/tutorials/tfx/gcp/vertex_pipelines_vertex_training#run_the_pipeline_using_a_gpu) and the model training code uses [tf.distribute.MirroredStrategy](https://www.tensorflow.org/api_docs/python/tf/distribute/MirroredStrategy). 
Hope this helps. Thank you!
